src/PerformanceManager.js

```diff
@@ -0,0 +1,178 @@
+const EventEmitter = require('events')
+const BigNumber = require('bignumber.js')
+
+class PerformanceManager extends EventEmitter {
+  /**
+   * @param priceFeed
+   * @param {BigNumber} maxPositionSize
+   * @param {BigNumber} allocation
+   */
+  constructor (priceFeed, {
+    maxPositionSize,
+    allocation
+  }) {
+    super()
+
+    this.maxPositionSize = maxPositionSize
+    this.allocation = allocation
+    this.availableFunds = new BigNumber(allocation)
+    this.priceFeed = priceFeed
+
+    this.peak = new BigNumber(allocation)
+    this.prevPeak = new BigNumber(0)
+    this.trough = new BigNumber(allocation)
+    this.openOrders = []
+
+    priceFeed.on('update', this.selfUpdate.bind(this))
+  }
+
+  /**
+   * @param {BigNumber} amount
+   * @param {BigNumber} price
+   * @returns {Error|null}
+   */
+  canOpenOrder (amount, price) {
+    const positionSize = this.positionSize()
+    const newPositionSize = positionSize.plus(amount)
+    if (newPositionSize.isGreaterThan(this.maxPositionSize)) {
+      return new Error(`order size exceeds maximum position size (order amount: ${amount}, current size: ${positionSize}, max size: ${this.maxPositionSize})`)
+    }
+
+    if (newPositionSize.isNegative()) {
+      return new Error('short positions are not allowed in this version')
+    }
+
+    const total = amount.multipliedBy(price)
+
+    const currentAllocation = this.currentAllocation()
+    const newAllocatedCapital = currentAllocation.plus(total)
+    if (newAllocatedCapital.isGreaterThan(this.allocation)) {
+      return new Error(`order exceeds max allocation (total: ${total}, current alloc: ${currentAllocation}, max alloc: ${this.allocation})`)
+    }
+
+    if (total.isGreaterThan(this.availableFunds)) {
+      return new Error(`order exceeds available funds (total: ${total}, available funds: ${this.availableFunds})`)
+    }
+
+    return null
+  }
+
+  /**
+   * @returns {BigNumber}
+   */
+  positionSize () {
+    return this.openOrders.reduce((size, order) =>
+      size.plus(order.amount),
+    new BigNumber(0)
+    )
+  }
+
+  /**
+   * @returns {BigNumber}
+   */
+  currentAllocation () {
+    return this.openOrders.reduce((alloc, order) =>
+      alloc.plus(order.amount.multipliedBy(order.price)),
+    new BigNumber(0)
+    )
+  }
+
+  addOrder ({ amount, price }) {
+    const total = amount.multipliedBy(price)
+
+    if (amount.isGreaterThanOrEqualTo(0)) {
```

Comment:
Seems redundant to handle case when amount is 0.
```
if(amount.isZero()) {
   return
}
```

Reply:
what I wanted to know if the value was positive, I've updated the code now to use the `isPositive` function